spec/features/consumer/shopping/checkout_spec.rb

```diff
@@ -433,5 +419,60 @@
         end.to enqueue_job ConfirmOrderJob
       end
     end
+
+    context "when the order cycle is modified before checkout is completed" do
+      before do
+        set_order order2
+        add_product_to_cart order2, product
+        add_product_to_cart order2, product2
+
+        # Remove one of the cart items from the OC
+        order_cycle2.exchanges.each do |e|
+          e.exchange_variants.first.delete
+        end
+
+        order_cycle2.save
+      end
+
+      it "redirects to cart page with message if items have been removed from the oc" do
+        visit checkout_path
+        checkout_as_guest
+
+        expect(order_cycle2.exchanges.outgoing.first.exchange_variants.count).to eq 1
+
+        fill_in_basic_checkout_details
+
+        place_order
+
+        expect(page).to have_content 'Your shopping cart'
+        expect(page).to have_content 'Items in your cart have been removed from the order cycle. Please review your order before proceeding.'
+      end
+    end
+  end
+
+  private
+
+  def fill_in_basic_checkout_details
```

Comment:
Did you check whether we have something like this or similar (I guess we do)? if so, I'd rather check if we can extend it or directly use it. If not, might be the time to move this new method into a helper module in `spec/support/` and start using it to DRY feature specs.